test/e2e/preflight_crdupgradesafety_scopechange_test.go

```diff
@@ -107,7 +107,7 @@ spec:
 `
 
 	update = strings.ReplaceAll(update, "__test-name__", testName)
-	logger.Section("deploy app with breaking change to CRD and preflight check enabled, should error", func() {
+	logger.Section("deploy app with CRD update that changes scope from namespace to cluster, preflight check enabled, should error", func() {
```

Comment:
Could this also be `cluster to namespace`?

Reply:
In theory yes that scope change could happen and it should be accurately captured. I wrote unit tests that capture the inverse so I figured a single e2e for namespace --> cluster is sufficient here.

Reply:
For reference: https://github.com/everettraven/kapp/blob/a0d4327d3b99df0753de20aaf662581ae9ca2108/pkg/kapp/crdupgradesafety/validator_test.go#L86-L99

Reply:
and from a technical perspective, the check only checks that the scope hasn't changed. You can change from cluster scoped to foobar scoped and it would also be rejected (although I'm not sure that is a valid change for Kubernetes :)).